src/xsar/sentinel1_meta.py

```diff
@@ -414,6 +308,25 @@ def geoloc(self):
             acq_atrack_meters, _ = haversine(*corners[1], *corners[2])
             self._geoloc.attrs['coverage'] = "%dkm * %dkm (atrack * xtrack )" % (
                 acq_atrack_meters / 1000, acq_xtrack_meters / 1000)
+            
+            # compute self._geoloc.attrs['approx_transform'], from gcps
+            # we need to convert self._geoloc to  a list of GroundControlPoint
+            def _to_rio_gcp(pt_geoloc):
+                # convert a point from self._geoloc grid to rasterio GroundControlPoint
+                return GroundControlPoint(
+                    x=pt_geoloc.longitude.item(),
+                    y=pt_geoloc.latitude.item(),
+                    z=pt_geoloc.altitude.item(),
+                    row=pt_geoloc.atrack.item(),
+                    col=pt_geoloc.xtrack.item()
+                )
+            rio_gcps = [
+                _to_rio_gcp(self._geoloc.sel(atrack=row, xtrack=col))
+                for row in  self._geoloc.atrack for col in self._geoloc.xtrack
+            ]
+            # approx transform, from all gcps (inaccurate)
+            approx_transform = rasterio.transform.from_gcps(rio_gcps)
+            self._geoloc.attrs['approx_transform'] = approx_transform
```

Comment:
Since `self._geoloc.attrs['approx_transform']` is now defined, you can probably try to clean more code

Reply:
Or perhaps all is already cleaned ? Do you think we are ready to merge @agrouaze ?

Reply:
Indeed I will define the `approx_transform`. The structure seems simple to replicate using `geoloc`:
![image](https://user-images.githubusercontent.com/14925067/162480378-45c7ae70-5a36-4550-8173-5c5ca4de8d30.png)